crates/cxx-qt/include/threading.h

```diff
@@ -32,11 +31,20 @@ class CxxQtThreading : public CxxQtLocking
 
   CxxQtThread<T> qtThread() const
   {
-    return CxxQtThread<T>(m_cxxQtThreadObj, m_rustObjMutex);
+    // Note that this assumes that T inherits from CxxQtLocking, which we assert
+    // during parsing.
+    return CxxQtThread<T>(m_cxxQtThreadObj);
   }
 
 private:
   ::std::shared_ptr<CxxQtGuardedPointer<T>> m_cxxQtThreadObj;
 };
 
+template<typename T>
+CxxQtThread<T>
+qtThread(const T& qobject)
+{
+  return static_cast<const CxxQtThreading<T>&>(qobject).qtThread();
```

Comment:
do we add a `Q_ASSERT` that the cast can work?

Reply:
It's a static cast, so it should fail to compile if the qobject doesn't inherit from CxxQtThreading.